pkg/composableschemadsl/compiler/importer.go

```diff
@@ -0,0 +1,51 @@
+package compiler
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/authzed/spicedb/pkg/composableschemadsl/input"
+	"github.com/authzed/spicedb/pkg/genutil/mapz"
+)
+
+type ImportContext struct {
+	pathSegments []string
+	sourceFolder string
+	names        *mapz.Set[string]
+}
+
+const SchemaFileSuffix = ".zed"
+
+func ImportFile(importContext *ImportContext) (*CompiledSchema, error) {
+	relativeFilepath := constructFilePath(importContext.pathSegments)
+	filepath := path.Join(importContext.sourceFolder, relativeFilepath)
+
+	var schemaBytes []byte
+	schemaBytes, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read schema file: %w", err)
+	}
+	// TODO: do we want this sort of logging here?
+	log.Trace().Str("schema", string(schemaBytes)).Str("file", filepath).Msg("read schema from file")
+
+	compiled, err := Compile(InputSchema{
```

Comment:
If you call an internal compile, then this could be passed the translationContext directly and then you wouldn't need a different context here or the need to pass it into Compile